### PR TITLE
Revert "TextField and last input character should visible on the screen when the cursor is not shown (#74722)"

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
** This is not a straightforward revert: ** https://github.com/flutter/flutter/pull/72828 changed a lot of what was in #74722 to resolve merge conflicts. The commit for resolving the merge conflicts: https://github.com/flutter/flutter/pull/72828/commits/c217f997c43d6d951303bb7cfdb7028e7e67d16a

This reverts commit cd771404e9da3606405c29e1c380684e449665c4 to unblock the roll.

### CODE REVIEW

**Review by justinmc — APPROVED** [review]
LGTM 👍 

This looks like what we need by my eye...  I've also submitted this to Frob to see if the Google tests pass.

**Comment by justinmc** [comment]
@xu-baolin A Google test was failing when your PR was merged.  It seemed to need a field to be deliberately focused instead of just calling `enterText`.  I think it was due to the fact that the PR changed the logic for when onCaretChanged was called.  Can you try to reopen a new PR that doesn't change that when you have time?

**Comment by xu-baolin** [comment]
thanks for revert.  I will rework on this.

**Review by xu-baolin — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/rendering/editable.dart [hunk 1]
```diff
@@ -2748,7 +2747,7 @@ class _FloatingCursorPainter extends RenderEditablePainter {
 
     final Color? floatingCursorColor = this.caretColor?.withOpacity(0.75);
     // Floating Cursor.
-    if (floatingCursorRect == null || floatingCursorColor == null || !shouldPaint)
+    if (floatingCursorRect == null || floatingCursorColor == null)
```
Comment (xu-baolin) [inline packages/flutter/lib/src/rendering/editable.dart:2750]: it seems like that this is not the original PR changed, am I missing something?

Comment (xu-baolin) [reply packages/flutter/lib/src/rendering/editable.dart:2750]: I understand now , it's the conflict PR logic.

Comment (LongCatIsLooong) [reply packages/flutter/lib/src/rendering/editable.dart:2750]: Yeah I had to resolve some merge conflicts in another RenderEditable PR, this is one of the places I had to change to accommodate the caret fix. See the description section

**Comment by LongCatIsLooong** [comment]
I think I have some clues about why it was failing in internal tests. A `tester.ensureVisible` call changed the layout and that triggered the `_showCaretOnScreen` callback of the currently focused text field. Not sure why `_textChangedSinceLastCaretUpdate` didn't prevent that from happening. I'll try to fix it and try running it on TGP.

**Comment by LongCatIsLooong** [comment]
@renyou should I disregard the tree closure and land the revert?

**Comment by xu-baolin** [comment]
Can someone googler help add a unit test to prevent it from failing again in the future?
also helpful to reland.

**Comment by renyou** [comment]
> @renyou should I disregard the tree closure and land the revert?

Yes, I think we can land this. It will help us to get out of the situation. Thanks!

**Review by renyou — APPROVED** [review]

**Pull request merged**